### PR TITLE
:bug: Fix hubspot datetime empty string

### LINKED ISSUE #5714: Hubspot Connector 0.1.11 Fails on Some Tables due to outputting empty strings as dates
## Enviroment
- **Airbyte version**:  0.29.12-alpha
- **OS Version / Instance**: Debian GNU/Linux 10 (buster) on GCP, e2-medium
- **Deployment**: Docker
- **Source Connector and version**: Hubspot 0.1.11
- **Destination Connector and version**: BigQuery 0.3.12
- **Severity**: High - Cannot currently sync **contacts**, **deals**, **companies**
- **Step where error happened**: Sync job

## Current Behavior
The following tables are not syncable from Hubspot

Failed On **SYNC**

- Deals
- Companies
- Contacts

**Normalization then also fails**

## Expected Behavior
All tables should sync and go through basic normalization

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```
2021-08-27 19:24:38 INFO () LineGobbler(voidCall):85 - 19:24:38 | 
2021-08-27 19:24:38 INFO () LineGobbler(voidCall):85 - 19:24:38 | Finished running 134 table models in 146.91s.
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - 
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - [31mCompleted with 3 errors and 0 warnings:[0m
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - 
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - [33mDatabase Error in model deals_properties (models/generated/airbyte_tables/airbyte_hubspot_test/deals_properties.sql)[0m
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 -   Invalid timestamp: ''
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/airbyte_hubspot_test/deals_properties.sql
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - 
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - [33mDatabase Error in model companies_properties (models/generated/airbyte_tables/airbyte_hubspot_test/companies_properties.sql)[0m
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 -   Invalid timestamp: ''
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/airbyte_hubspot_test/companies_properties.sql
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - 
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - [33mDatabase Error in model contacts_properties (models/generated/airbyte_tables/airbyte_hubspot_test/contacts_properties.sql)[0m
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 -   Invalid timestamp: ''
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 -   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_tables/airbyte_hubspot_test/contacts_properties.sql
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - 
2021-08-27 19:24:40 INFO () LineGobbler(voidCall):85 - Done. PASS=131 WARN=0 ERROR=3 SKIP=0 TOTAL=134
2021-08-27 19:24:41 INFO () TemporalAttemptExecution(lambda$getWorkerThread$2):149 - Completing future exceptionally...
io.airbyte.workers.WorkerException: Normalization Failed.
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:69) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: io.airbyte.workers.WorkerException: Normalization Failed.
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:66) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	... 3 more
	Suppressed: io.airbyte.workers.WorkerException: Normalization process wasn't successful
		at io.airbyte.workers.normalization.DefaultNormalizationRunner.close(DefaultNormalizationRunner.java:151) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
		at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:60) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
		at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
		at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
		at java.lang.Thread.run(Thread.java:832) [?:?]
2021-08-27 19:24:41 INFO () TemporalAttemptExecution(get):133 - Stopping cancellation check scheduling...
2021-08-27 19:24:41 WARN () POJOActivityTaskHandler$POJOActivityImplementation(execute):243 - Activity failure. ActivityId=9f2db00f-d115-3d21-a97b-54e2c9854484, activityType=Normalize, attempt=1
java.util.concurrent.ExecutionException: io.airbyte.workers.WorkerException: Normalization Failed.
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2063) ~[?:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.get(TemporalAttemptExecution.java:131) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at io.airbyte.workers.temporal.SyncWorkflow$NormalizationActivityImpl.normalize(SyncWorkflow.java:291) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:564) ~[?:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityInboundCallsInterceptor.execute(POJOActivityTaskHandler.java:277) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityImplementation.execute(POJOActivityTaskHandler.java:216) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler.handle(POJOActivityTaskHandler.java:181) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:192) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:154) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.PollTaskExecutor.lambda$process$0(PollTaskExecutor.java:73) ~[temporal-sdk-1.0.4.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: io.airbyte.workers.WorkerException: Normalization Failed.
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:69) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	... 1 more
Caused by: io.airbyte.workers.WorkerException: Normalization Failed.
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:66) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
	... 1 more
	Suppressed: io.airbyte.workers.WorkerException: Normalization process wasn't successful
		at io.airbyte.workers.normalization.DefaultNormalizationRunner.close(DefaultNormalizationRunner.java:151) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
		at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:60) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
		at io.airbyte.workers.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:36) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
		at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:146) ~[io.airbyte-airbyte-workers-0.29.12-alpha.jar:?]
		at java.lang.Thread.run(Thread.java:832) [?:?]

```

</details>

## Steps to Reproduce
1.  `wget https://raw.githubusercontent.com/airbytehq/airbyte/v0.29.12-alpha/{.env,docker-compose.yaml}`
2. `docker-compose up`
3. Update Hubspot connector to 0.1.11
4. select `companies`, `contacts`, `deals`, `deal_pipelines` (deal_pipelines seems to be fine)
5. run sync

## Are you willing to submit a PR?

Someday

### PR DESCRIPTION
**Pull request by avida**
## What

Fix for  #5714 

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by avida** [comment]
\test connectors=source-hubspot

File: buildSrc/src/main/groovy/airbyte-source-acceptance-test.gradle [hunk 1]
```diff
@@ -19,7 +19,7 @@ class AirbyteSourceAcceptanceTestPlugin implements Plugin<Project> {
                         '-w', "$targetMountDirectory",
                         '-e', "AIRBYTE_SAT_CONNECTOR_DIR=${project.projectDir.absolutePath}",
                         'airbyte/source-acceptance-test',
-                        '-p', 'integration_tests.acceptance',
+                        '-p', 'source_acceptance_test.plugin',
```
Comment (yevhenii-ldv) [inline buildSrc/src/main/groovy/airbyte-source-acceptance-test.gradle:22]: I don’t understand a little, a change in this line leads to the fact that we can delete the acceptance.py and integration_test.py files, but still use the SAT tests?

File: airbyte-integrations/bases/base-python/setup.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
         "PyYAML==5.4",
         "pydantic==1.6.*",
         "airbyte-protocol",
-        "jsonschema==2.6.0",
+        "jsonschema==3.2.0",
```
Comment (yevhenii-ldv) [inline airbyte-integrations/bases/base-python/setup.py:40]: Are you sure that `jsonschema `version 3 is backward compatible with version 2 and this change will not generate new bugs?

File: buildSrc/src/main/groovy/airbyte-source-acceptance-test.gradle (hunk 1 above)
Comment (avida) [reply buildSrc/src/main/groovy/airbyte-source-acceptance-test.gradle:22]: yes, you don't need acceptance.py, all test cases will be picked up by plugin according to acceptance test config

File: airbyte-integrations/bases/base-python/setup.py (hunk 2 above)
Comment (avida) [reply airbyte-integrations/bases/base-python/setup.py:40]: not 100% sure but acceptance test passed fine. There were dependencies conflicts with SAT that failed a build so Ive made this change. AFAIK base-python is outdated and we should move away from it.

**Review by yevhenii-ldv — APPROVED** [review]
LGTM, please, resolve conflicts

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/asserts.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def check_datetime(value: str) -> bool:
         return valid_format and valid_time
 
     def check(self, instance, format):
-        if format == "date-time":
+        if instance is not None and format == "date-time":
```
Comment (sherifnada) [inline airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/asserts.py:55]: make sure to publish SAT

**Review by sherifnada — APPROVED** [review]

**Comment by avida** [comment]
Cannot publish cause github runner' ubuntu-latest image have python with setuptools defect: https://github.com/pypa/setuptools_scm/issues/608 Will try later when image will be updated (hopefully)

**Pull request merged**